File: 874-backspace-string-compare/backspace-string-compare.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var backspaceCompare = function (s, t) {
    let S_stack = []
    let t_stack = []
    for (let i = 0; i < s.length; i++) {
        S_stack.push(s[i])
        if (s[i] == "#") {
            S_stack.pop()
            S_stack.pop()
        }
    }
    for (let i = 0; i < t.length; i++) {
        t_stack.push(t[i])
        if (t[i] == "#") {
            t_stack.pop()
            t_stack.pop()
        }
    }
    return S_stack.join("") == t_stack.join("")
};

File: 874-backspace-string-compare/backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        s_stack = []
        t_stack = []
        for i in range(len(s)):
            s_stack.append(s[i])
            if(s[i] == "#"):
                s_stack.pop()
                if s_stack:
                    s_stack.pop()
        for i in range(len(t)):
            t_stack.append(t[i])
            if(t[i] == "#"):
                t_stack.pop()
                if t_stack:
                    t_stack.pop()
        return "".join(s_stack) == "".join(t_stack)